source/location.go

```diff
@@ -173,8 +173,25 @@ type ArgumentOption struct {
 }
 
 type MessageValidationOption struct {
-	Idx  int
-	Rule bool
+	Idx    int
+	Rule   bool
+	Detail *MessageValidationDetailOption
+}
+
+type MessageValidationDetailOption struct {
+	Idx                 int
+	Rule                bool
+	PreconditionFailure *MessageValidationDetailPreconditionFailureOption
+}
+
+type MessageValidationDetailPreconditionFailureOption struct {
+	Idx       int
+	Violation MessageValidationDetailPreconditionFailureViolationOption
+}
+
+type MessageValidationDetailPreconditionFailureViolationOption struct {
+	Idx       int
+	FieldName string
```

Comment:
This is actually slightly different from the other options, but I believe having the field name directly is more concise than creating boolean fields for each field in the PreconditionFailure_Violation in this case. This works fine since all the fields are strings.